pkg/master/storage_rbac.go

```diff
@@ -91,3 +98,42 @@ func (p RBACRESTStorageProvider) v1alpha1Storage(apiResourceConfigSource generic
 	}
 	return storage
 }
+
+func (p RBACRESTStorageProvider) PostStartHook(apiResourceConfigSource genericapiserver.APIResourceConfigSource, restOptionsGetter RESTOptionsGetter) func() error {
+	var directClusterRoleAccess *clusterroleetcd.REST
+
+	if apiResourceConfigSource.AnyResourcesForVersionEnabled(rbacapiv1alpha1.SchemeGroupVersion) {
+		if apiResourceConfigSource.ResourceEnabled(rbacapiv1alpha1.SchemeGroupVersion.WithResource("clusterroles")) {
+			directClusterRoleAccess = clusterroleetcd.NewREST(restOptionsGetter(rbac.Resource("clusterroles")))
+		}
+	}
+
+	if directClusterRoleAccess == nil {
+		return func() error { return nil }
+	}
+
+	return func() error {
+		ctx := api.NewContext()
+
+		existingClusterRoles, err := directClusterRoleAccess.List(ctx, &api.ListOptions{})
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("unable to initialize clusterroles: %v", err))
+			return nil
+		}
+		// if clusterroles already exist, then assume we don't have work to do
```

Comment:
> Wait, you reject everything not from self until the post starts have been called? I did not see where that happened in this PR?

It's not new in this pull. RBAC was initially merged with an arg for a "magic" user who can do anything.  Everyone else was denied until other roles and bindings were created.  My plan is to automate the creation of cluster roles and  change the magic RBAC user to actually bind to a `ClusterRole` instead of being special cased.

That will start collapsing the authz paths that we have in the authorizer and provide a location to start adding a reasonable set of bootstrap roles.